plugin/metadata/metadata.go

```diff
@@ -0,0 +1,48 @@
+package metadata
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+// Metadater interface needs to be implemented by each plugin willing to provide
+// metadata information for other plugins.
+// Note: this method should work quickly, because it is called for every request
+// from the metadata plugin.
+type Metadater interface {
```

Comment:
It's awkward but I think this needs to be "Metadataer" - of if that is just too ugly we can do "MetadataProvider" or something like that. But "Metadater" is not right from an English perspective.